pypsrp/client.py

```diff
@@ -91,9 +94,25 @@ def read_buffer(b_path, total_size, buffer_size):
                 if offset == 0:
                     yield [u"", to_unicode(base64.b64encode(to_bytes(sha1.hexdigest())))]
 
+        def expand_path(path):
+            if expand_variables not in ("cmd", "ps"):
+                return path
+            if expand_variables == "cmd":
+                out, err, failed = self.execute_cmd("echo {}".format(path))
+                out = out.rstrip()
+            else:
+                out, err, failed = self.execute_ps("Write-Output {}".format(path))
```

Comment:
Once we decide on the way forward we should limit calls to `execute_*` in this and just reuse the existing RunspacePool we open below. This reduces the amount of roundtrips we take to the server which can be an expensive operation in some environments. Usually this is as simple as creating separate statements and storing a variable so each statement can access the information.

Reply:
yes I agree. I feel like I will have to do some refactoring here because the function will be too big if I put more stuff into the "runspace section". So I decided not to change stuff around when I felt like I will rewrite it anyway

Reply:
I'm not 100% sure what you mean by the function will be too big?

Reply:
too many lines of code :)